utils/build_swift/build_swift/driver_arguments.py

```diff
@@ -584,6 +584,10 @@ def create_argument_parser():
            help="When building on Darwin, automatically populate cross-compile-hosts "
                 "based on the architecture build-script is running on. "
                 "Has precedence over cross-compile-hosts")
+    # This is meant to test --infer-cross-compile-hosts-on-darwin
+    # in dry-run mode, so we don't want to advertise this to most users
+    option('--dry-run-assume-host-arch-is', store,
+           help=argparse.SUPPRESS)
```

Comment:
Given that we don't want to make this generally available and that it is specifically for a singular test ATM, what do you think of honoring an environment variable (e.g. `ARCH`) instead of having a dedicated option that we then have to explicitly suppress?  The changes to `build-script` are also then simpler.

Reply:
That makes a lot of sense, I had in mind the hidden options we have on the frontend and did not think about the big picture.

I took this feedback in e0395745d6a8f859b8bc0092e81bfbf0bb4b04e7 -- I am still ensuring to sense the variable in dry run and print when such sensing occurring in a effort to reduce the likelihood of using this code path in normal usage.